File: levelledmobs-plugin-bukkit/src/main/java/io/github/arcaneplugins/levelledmobs/bukkit/listener/impl/EntityBreedListener.java
package io.github.arcaneplugins.levelledmobs.bukkit.listener.impl;

import io.github.arcaneplugins.levelledmobs.bukkit.data.InternalEntityDataUtil;
import io.github.arcaneplugins.levelledmobs.bukkit.listener.ListenerWrapper;
import io.github.arcaneplugins.levelledmobs.bukkit.logic.LogicHandler;
import io.github.arcaneplugins.levelledmobs.bukkit.logic.context.Context;
import org.bukkit.entity.LivingEntity;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.entity.EntityBreedEvent;

public class EntityBreedListener extends ListenerWrapper {

    public EntityBreedListener() {
        super(true);
    }

    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
    public void handle(final EntityBreedEvent event) {
        final LivingEntity child = event.getEntity();
        final LivingEntity father = event.getFather();
        final LivingEntity mother = event.getMother();

        InternalEntityDataUtil.setFather(child, father, true);
        InternalEntityDataUtil.setMother(child, mother, true);
        InternalEntityDataUtil.setWasBred(child, true, true);

        LogicHandler.runFunctionsWithTriggers(
            new Context()
                .withEntity(child)
                .withFather(father)
                .withMother(mother)
            , "on-entity-breed"
        );
    }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/bukkit/listener/impl/EntityBreedListener.kt
package io.github.arcaneplugins.levelledmobs.bukkit.listener.impl

import io.github.arcaneplugins.levelledmobs.bukkit.data.InternalEntityDataUtil.setFather
import io.github.arcaneplugins.levelledmobs.bukkit.data.InternalEntityDataUtil.setMother
import io.github.arcaneplugins.levelledmobs.bukkit.data.InternalEntityDataUtil.setWasBred
import io.github.arcaneplugins.levelledmobs.bukkit.listener.ListenerWrapper
import io.github.arcaneplugins.levelledmobs.bukkit.logic.LogicHandler.runFunctionsWithTriggers
import io.github.arcaneplugins.levelledmobs.bukkit.logic.context.Context
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.entity.EntityBreedEvent

class EntityBreedListener : ListenerWrapper(true) {
    @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
    fun handle(event: EntityBreedEvent){
        val child = event.entity
        val father = event.father
        val mother = event.mother

        setFather(child, father, true)
        setMother(child, mother, true)
        setWasBred(child, true, true)

        runFunctionsWithTriggers(
            Context()
                .withEntity(child)
                .withFather(father)
                .withMother(mother)
            , mutableListOf("on-entity-breed")
        )
    }
}